htlcswitch/link.go

```diff
@@ -3085,6 +3099,25 @@ func (l *channelLink) processRemoteAdds(fwdPkg *channeldb.FwdPkg,
 
 		fwdInfo := pld.ForwardingInfo()
 
+		// If we are in a flush state we need to cancel back all of the
+		// net new HTLCs rather than forwarding them. This is the first
+		// opportunity we have to bounce invalid HTLC adds without
+		// doing a force-close.
+		if l.IsFlushing() {
```

Comment:
👍

Reply:
We should only do this cancel back logic if the state of `fwdPkg` is `FwdStateLockedIn`. If the state is `FwdStateProcessed`, the htlc may already have been sent to the outgoing link in a prior invocation of `processRemoteAdds` and canceling back would be unsafe